datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/counter.py

```diff
@@ -0,0 +1,516 @@
+        # https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhremovecounter
+        # http://timgolden.me.uk/pywin32-docs/win32pdh__RemoveCounter_meth.html
+        win32pdh.RemoveCounter(self.counter_handle)
+        self.counter_handle = None
+
+
+class MultiCounter(CounterBase):
+    def __init__(
+        self,
+        name,
+        config,
+        check,
+        connection,
+        object_name,
+        metric_prefix,
+        counter_selector,
+        tags,
+        custom_transformer,
+        tag_name,
+    ):
+        super().__init__(
+            name, config, check, connection, object_name, metric_prefix, counter_selector, tags, custom_transformer
+        )
+
+        self.tag_name = tag_name
+
+        self.average = config.get('average', False)
+        if not isinstance(self.average, bool):
+            raise ConfigTypeError(
+                f'Option `average` for counter `{self.name}` of performance '
+                f'object `{self.object_name}` must be a boolean'
+            )
+
+        self.aggregate_transformer = None
+        self.aggregate = config.get('aggregate', False)
+        if not isinstance(self.aggregate, bool) and self.aggregate != 'only':
+            raise ConfigTypeError(
+                f'Option `aggregate` for counter `{self.name}` of performance '
+                f'object `{self.object_name}` must be a boolean or set to `only`'
+            )
+        elif self.aggregate is not False:
+            if self.metric_type not in NATIVE_TRANSFORMERS:
+                raise ConfigTypeError(
+                    f'Option `aggregate` for counter `{self.name}` of performance object `{self.object_name}` is '
+                    f'enabled so `type` must be set to one of the following: {", ".join(NATIVE_TRANSFORMERS)}'
+                )
+            else:
+                metric_name = f'{self.metric_name}.avg' if self.average else f'{self.metric_name}.sum'
+                self.aggregate_transformer = NATIVE_TRANSFORMERS[self.metric_type](check, metric_name, config)
+
+        # All monitored counter handles keyed by the instance name
+        self.instances = {}
+
+    def collect(self):
+        handles_with_data = 0
+        total = 0
+        for instance, counter_handles in self.instances.items():
+            # Some instances may not have a value yet, so we only
+            # use the ones that do for computing averages
+            instance_handles_with_data = len(counter_handles)
+            instance_total = 0
+
+            for i, counter_handle in enumerate(counter_handles):
+                try:
+                    value = get_counter_value(counter_handle)
+                except pywintypes.error as error:
+                    instance_handles_with_data -= 1
+                    self.handle_counter_value_error(error, instance if i == 0 else f'{instance}#{i}')
+                else:
+                    instance_total += value
+
+            if not instance_handles_with_data:
```

Comment:
```suggestion
            if instance_handles_with_data <= 0:
```
small nit for readability

Reply:
hmm, since it can never be less than zero then this is a check for zero so this already seems idiomatic